### PR TITLE
Major bug fix and some optimizations

### PR DESCRIPTION
**Pull request by GuillermoCasas**
Now the search_increment acts as expected from the benchmarks. Next step is integrating all this into SpreadDEM

### CODE REVIEW

**Review by maceligueta — APPROVED** [review]

File: applications/DEM_application/test_examples/basic_benchmarks/BenchTemp.info [hunk 1]
```diff
@@ -0,0 +1,11246 @@
...
+Assigning ForwardEulerScheme to properties 1
+          **************************************************
+            Parallelism Info:  MPI number of nodes: 1
+            Parallelism Info:  OMP number of processors: 1
+          **************************************************
+
+Initialization Complete
+
+Main loop starting...
+Total number of TIME S
```
Comment (roigcarlo) [inline applications/DEM_application/test_examples/basic_benchmarks/BenchTemp.info]: Is this file really needed?

Comment (farrufat-cimne) [reply applications/DEM_application/test_examples/basic_benchmarks/BenchTemp.info]: it is not.

Comment (GuillermoCasas) [reply applications/DEM_application/test_examples/basic_benchmarks/BenchTemp.info]: I have removed the file. Sorry, it slipped.

**Comment by maceligueta** [comment]
Apparently, the time needed for compilation is exceeding the available time. I'm not sure we will manage to merge this.

**Comment by roigcarlo** [comment]
We are workning on it in several fronts, in the meantime rebuilding the commit will most likely trigger the cache

**Comment by philbucher** [comment]
Can this branch be deleted?
